COOKBOOK.md

```diff
@@ -0,0 +1,964 @@
+  tableRowLast: ViewStyle;
+  text: TextStyle;
+  u: TextStyle;
+  view: ViewStyle;
+}>;
+```
+
+### Message with custom reactions
+
+To add custom reactions you need to use the `supportedReactions` prop on `Channel`. `supportedReactions` is an array of `ReactionData`. The default `supportedReactions` array contains 5 reactions.
+
+```tsx
+export const reactionData: ReactionData[] = [
+  {
+    Icon: LoveReaction,
+    type: 'love',
+  },
+  {
+    Icon: ThumbsUpReaction,
+    type: 'like',
+  },
+  {
+    Icon: ThumbsDownReaction,
+    type: 'sad',
+  },
+  {
+    Icon: LOLReaction,
+    type: 'haha',
+  },
+  {
+    Icon: WutReaction,
+    type: 'wow',
+  },
+];
+```
+
+To create your own reaction you need both a `type` and `Icon`. The `Icon` is a component with `IconProps` it is suggested you take advantage of [`react-native-svg`](https://github.com/react-native-svg/react-native-svg) for scaling purposes. It is suggested you look at the default icons for examples of how to create your own that is able to properly use the theme and sizing that are provided via props. Using exported type from `stream-chat-react-native` a custom reaction can be created and added.
+
+```tsx
+export const StreamReaction: React.FC<IconProps> = (props) => (
+  <RootSvg height={21} width={42} {...props} viewBox='0 0 42 21'>
+    <RootPath
+      d='M26.1491984,6.42806971 L38.9522984,5.52046971 C39.7973984,5.46056971 40.3294984,6.41296971 39.8353984,7.10116971 L30.8790984,19.5763697 C30.6912984,19.8379697 30.3888984,19.9931697 30.0667984,19.9931697 L9.98229842,19.9931697 C9.66069842,19.9931697 9.35869842,19.8384697 9.17069842,19.5773697 L0.190598415,7.10216971 C-0.304701585,6.41406971 0.227398415,5.46036971 1.07319842,5.52046971 L13.8372984,6.42816971 L19.2889984,0.333269706 C19.6884984,-0.113330294 20.3884984,-0.110730294 20.7846984,0.338969706 L26.1491984,6.42806971 Z M28.8303984,18.0152734 L20.5212984,14.9099734 L20.5212984,18.0152734 L28.8303984,18.0152734 Z M19.5212984,18.0152734 L19.5212984,14.9099734 L11.2121984,18.0152734 L19.5212984,18.0152734 Z M18.5624984,14.1681697 L10.0729984,17.3371697 L3.82739842,8.65556971 L18.5624984,14.1681697 Z M21.4627984,14.1681697 L29.9522984,17.3371697 L36.1978984,8.65556971 L21.4627984,14.1681697 Z M19.5292984,13.4435697 L19.5292984,2.99476971 L12.5878984,10.8305697 L19.5292984,13.4435697 Z M20.5212984,13.4435697 L20.5212984,2.99606971 L27.4627984,10.8305697 L20.5212984,13.4435697 Z M10.5522984,10.1082697 L12.1493984,8.31366971 L4.34669842,7.75446971 L10.5522984,10.1082697 Z M29.4148984,10.1082697 L27.8178984,8.31366971 L35.6205984,7.75446971 L29.4148984,10.1082697 Z'
+      {...props}
+    />
+  </RootSvg>
+);
+
+const newReactionData = [...reactionData, { type: 'stream', Icon: StreamReaction }];
+```
+
+Both the resulting reaction picker and reaction result can then utilize this additional option.
+
+<table>
+  <tr>
+    <td align='center'><img src='./screenshots/cookbook/StandardReactions.png' width="300"/></td>
```

Comment:
should we be doing percentage widths for these tables? `width="300"` -> `width="33%"`?

Reply:
Does that work on github? They strip a lot on the render, if it works then 100% we should.

Reply:
honestly not sure? I can try it on another project but I'd assume it would work like markdown should

Reply:
html is HEAVILY stripped on github is the possible issue, I can test it here in a minute.